From 095396d9ea4cfc3b6777935b64f68f0693676c94 Mon Sep 17 00:00:00 2001
From: Kyle Stein <kyle.stein@arm.com>
Date: Wed, 25 Mar 2020 09:14:14 -0500
Subject: [PATCH 06/14] Call a script on factory reset

Add a hook to a script on factory reset.  This gives us a chance
to delete persistent customer logs from the gateway.

Signed-off-by: Kyle Stein <kyle.stein@arm.com>
---
 edge-core/edge_server_customer_code.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/edge-core/edge_server_customer_code.c b/edge-core/edge_server_customer_code.c
index 4ab25a0..bdc3908 100644
--- a/edge-core/edge_server_customer_code.c
+++ b/edge-core/edge_server_customer_code.c
@@ -21,6 +21,7 @@
 #include "edge-client/edge_client.h"
 #include "mbed-trace/mbed_trace.h"
 #include "edge-core/edge_server_customer_code.h"
+#include <stdlib.h>
 #define TRACE_GROUP "escstmr"
 
 bool edgeserver_execute_rfs_customer_code(edgeclient_request_context_t *request_ctx)
@@ -29,6 +30,15 @@ bool edgeserver_execute_rfs_customer_code(edgeclient_request_context_t *request_
             request_ctx->object_id,
             request_ctx->object_instance_id,
             request_ctx->resource_id);
+
+    // Execute a script to do factory reset tasks, such as clearing customer logs.
+    int rc = system("edge-core-factory-reset");
+
+    if (rc) {
+        tr_warn("edge-core-factory-reset exited with non-success return code %d", rc);
+        return false;
+    }
+
     return true;
 }
 
-- 
2.17.1

